nextcord/automod.py

```diff
@@ -0,0 +1,370 @@
+        HTTPException
+            Editing the rule failed.
+        Forbidden
+            You do not have proper permissions to edit this rule.
+        """
+        payload = {}
+        if name is not MISSING:
+            payload["name"] = name
+
+        if event_type is not MISSING:
+            payload["event_type"] = event_type.value
+
+        if keyword_filters is not MISSING and presets is not MISSING:
+            raise InvalidArgument("Cannot pass keyword_filters and presets to edit()")
+
+        if keyword_filters is not MISSING and self.trigger_type != TriggerType.keyword:
+            raise InvalidArgument(
+                "trigger_type must be TriggerType.keyword to pass keyword_filters"
+            )
+
+        if keyword_filters is not MISSING:
+            payload["trigger_metadata"]["keyword_filter"] = keyword_filters
+
+        if presets is not MISSING and self.trigger_type != TriggerType.keyword_presets:
+            raise InvalidArgument(
+                "trigger_type must be TriggerType.keyword_presets to pass presets"
+            )
+
+        if presets is not MISSING:
+            payload["trigger_metadata"]["presets"] = [preset.value for preset in presets]
+
+        if (
+            notify_channel is not MISSING
+            or timeout_seconds is not MISSING
+            or block_message is not MISSING
+        ):
+            payload["actions"] = []
+            if block_message is not MISSING:
+                if block_message:
+                    payload["actions"].append({"type": 1})
+            else:
+                for action in self.actions:
+                    if action.type is ActionType.block:
+                        payload["actions"].append({"type": 1})
+
+        if (notify_channel is not MISSING) and (timeout_seconds is MISSING):
+            payload["actions"].append({"type": 2, "metadata": {"channel_id": notify_channel.id}})
+            for action in self.actions:
+                if action.timeout_seconds is not None:
+                    payload["actions"].append(
+                        {"type": 3, "metadata": {"duration_seconds": action.timeout_seconds}}
+                    )
+
+        if (timeout_seconds is not MISSING) and (notify_channel is MISSING):
+            payload["actions"].append(
+                {"type": 3, "metadata": {"duration_seconds": timeout_seconds}}
+            )
+            for action in self.actions:
+                if action.notify_channel_id is not None:
+                    payload["actions"].append(
+                        {"type": 2, "metadata": {"channel_id": action.notify_channel_id}}
+                    )
+
+        if timeout_seconds is not MISSING) and (notify_channel is not MISSING:
+            payload["actions"].append({"type": 2, "metadata": {"channel_id": notify_channel.id}})
+            payload["actions"].append(
+                {"type": 3, "metadata": {"duration_seconds": timeout_seconds}}
+            )
+
```

Comment:
```suggestion

        if (
            notify_channel is not MISSING
            or timeout_seconds is not MISSING
            or block_message is not MISSING
        ):
            payload["actions"] = []
            if block_message is not MISSING:
                if block_message:
                    payload["actions"].append({"type": 1})
            else:
                for action in self.actions:
                    if action.type is ActionType.block:
                        payload["actions"].append({"type": 1})

        if notify_channel is not MISSING and timeout_seconds is MISSING:
            payload["actions"].append({"type": 2, "metadata": {"channel_id": notify_channel.id}})
            for action in self.actions:
                if action.timeout_seconds is not None:
                    payload["actions"].append(
                        {"type": 3, "metadata": {"duration_seconds": action.timeout_seconds}}
                    )

        if timeout_seconds is not MISSING and notify_channel is MISSING:
            payload["actions"].append(
                {"type": 3, "metadata": {"duration_seconds": timeout_seconds}}
            )
            for action in self.actions:
                if action.notify_channel_id is not None:
                    payload["actions"].append(
                        {"type": 2, "metadata": {"channel_id": action.notify_channel_id}}
                    )

        if timeout_seconds is not MISSING and notify_channel is not MISSING:
            payload["actions"].append({"type": 2, "metadata": {"channel_id": notify_channel.id}})
            payload["actions"].append(
                {"type": 3, "metadata": {"duration_seconds": timeout_seconds}}
            )

```

Reply:
i'm too lazy to open the ide to fix shit so i work by suggesting code for myself lol

Reply:
You do realize that you are not working in your private repository and nextcord is meant to be something proper which would involve some form of professionalism?

Reply:
sorry